reseq_ros2/reseq_ros2/communication.py

```diff
@@ -14,7 +14,7 @@ class Communication(Node):
     def __init__(self):
         super().__init__("communication")
         # TODO: read file passed as ROS argument
-        with open("src/reseq_ros2/reseq_ros2/config.yaml") as f:
+        with open("src/reseq_ros2/reseq_ros2/config_mk2.yaml") as f:
```

Comment:
Having 2 (4) separate configuration files for mark 1 and 2 is not ideal, as we would have to edit the code every time, even if it's just a couple of lines.

I would put every constant (including the ones mentioned in #2) inside a single yaml file for both versions of the robot. Something like this: 
```
mk1:
  constants:
    a: 0.18
    b: 0.21
  modules:
  - address: 17
    hasJoint: false
    hasEndEffector: false
  - address: 18
    hasJoint: false
    hasEndEffector: false
mk2:
  constants:
    a: 0.48
    b: 0.37
  modules:
  - address: 27
    hasJoint: true
    hasEndEffector: false
  - address: 28
    hasJoint: false
    hasEndEffector: false
```
Then we can use a ROS [parameter](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Using-Parameters-In-A-Class-Python.html) (a simple integer) as suggested by @marc0777 to choose between `mk1` and `mk2`. That way we can change it from the command line or a launch file without having to touch the code.

Reply:
I agree that this is far from ideal, however I like the idea of having more than one configuration file, and pass as parameter the one representing the robot we have

Reply:
I'm fine with the idea of having separate config files as long as we can choose which one needs to be used without touching the Python code. I'll take a look at how ROS parameters and arguments work

https://docs.ros.org/en/humble/How-To-Guides/Node-arguments.html
https://docs.ros.org/en/humble/How-To-Guides/Parameters-YAML-files-migration-guide.html

Reply:
So @GBergatto are you working on it or should I go ahead?

Reply:
Yes, I can work on the config files, while you handle vel_motors

Reply:
I just pushed a commit setting up the work on parameters, sorry if I stole some of your work, I had already started before this conversation...

Reply:
Something similar should be done on communication, and then the parameters should be all passed from a launch file, that can read the yaml file